library/test/src/test/java/com/bumptech/glide/load/model/StreamEncoderTest.java

```diff
@@ -47,4 +49,38 @@ public void testWritesDataFromInputStreamToOutputStream() throws IOException {
 
     assertEquals(fakeData, new String(data, "UTF-8"));
   }
+
+  @Test(expected = NullPointerException.class)
+  public void testFileNullException() throws IOException {
+    String fakeData = "SomeRandomFakeData";
+    ByteArrayInputStream is = new ByteArrayInputStream(fakeData.getBytes("UTF-8"));
+    boolean success = false;
+
+    // Write some data to #file making sure file.delete() executes properly
+    encoder.encode(is, file, new Options());
+
+    success = encoder.encode(is, null, new Options());
+    // method should end here, since null file exception
+
+    // If #encoder.encode() does not throw exception, encoding process should be successful, assert true
+    assertTrue(success);
+  }
+
+  @Test
+  public void testImportanceOfOptionsArgument() throws IOException {
```

Comment:
Can you rename this to: `encode_withNullOptions_completesSuccessfully`?

Reply:
Done